### ISSUE TITLE
SQL: TIME and DATE types metadata attributes corrections

### ISSUE THREAD

**Issue by bpintea** [elastic/elasticsearch#41086]
The values returned by `SYS TYPES` for the time data types needs adjusting for the following attributes: `MINIMUM_SCALE`, `MAXIMUM_SCALE` and `SQL_DATETIME_SUB` as follows:
* the `DATE` and `TIME` must have the values swapped for `MINIMUM_SCALE` and `MAXIMUM_SCALE` attributes (since `DATE` has no seconds component);
* `SQL_DATETIME_SUB` must have these subcode values:
  * for `DATE`: `1`;
  * for `TIME`: `2`.

Current values (for convenience):
```
        TYPE_NAME        |   DATA_TYPE   |   PRECISION   |LITERAL_PREFIX |LITERAL_SUFFIX | CREATE_PARAMS |   NULLABLE    |CASE_SENSITIVE |  SEARCHABLE   |UNSIGNED_ATTRIBUTE|FIXED_PREC_SCALE|AUTO_INCREMENT |LOCAL_TYPE_NAME| MINIMUM_SCALE | MAXIMUM_SCALE | SQL_DATA_TYPE |SQL_DATETIME_SUB|NUM_PREC_RADIX |INTERVAL_PRECISION
-------------------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+---------------+------------------+----------------+---------------+---------------+---------------+---------------+---------------+----------------+---------------+------------------
DATE                     |91             |29             |'              |'              |null           |2              |false          |3              |true              |false           |false          |null           |3              |3              |91             |0               |null           |null              
TIME                     |92             |18             |'              |'              |null           |2              |false          |3              |true              |false           |false          |null           |null           |null           |92             |0               |null           |null              
DATETIME                 |93             |29             |'              |'              |null           |2              |false          |3              |true              |false           |false          |null           |3              |3              |9              |3               |null           |null              

```

**Comment by ithompson4** [comment]
Could I work on this? Thanks.

**Comment by matriv** [comment]
@ithompson4 Feel free to work on it. The class to start with is: https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypes.java

**Comment by matriv** [comment]
Fixed by #55316

**Comment by matriv** [comment]
`master` : c23677cd2955e25bb952c8e7ff8ca3151ee0df98
`7.x` : 8abdf7c7d31f50dbb12f3e59034fd83c040166d4
`7.7` : 736b8a8030b732373b4069b3cdca9fa2ec89e72b
